crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2344,6 +2344,352 @@ pub fn local_equivalence(weyl: PyReadonlyArray1<f64>) -> PyResult<[f64; 3]> {
+            if gamma > 0.0 {
+                gamma *= -1.0;
+                invert = true;
+            }
+            let circ_c = self.to_rxx_gate(gamma)?;
+            if !invert {
+                global_phase += circ_c.global_phase;
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+                circ.gates.extend(circ_c.gates);
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+            } else {
+                // invert the circuit above
+                global_phase -= circ_c.global_phase;
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+                for gate in circ_c.gates.into_iter().rev() {
+                    let (inv_gate_name, inv_gate_params, inv_gate_qubits) = invert_2q_gate(gate);
+                    circ.gates
+                        .push((Some(inv_gate_name), inv_gate_params, inv_gate_qubits));
+                }
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![0]));
+                circ.gates
+                    .push((Some(StandardGate::HGate), smallvec![], smallvec![1]));
+            }
+        }
+
+        circ.global_phase = global_phase;
+        Ok(())
+    }
+
+    ///  Returns the Weyl decomposition in circuit form.
+    ///  Note: atol is passed to OneQubitEulerDecomposer.
+    fn call_inner(
+        &self,
+        unitary: ArrayView2<Complex64>,
+        atol: f64,
+    ) -> PyResult<TwoQubitGateSequence> {
+        let target_decomposed =
+            TwoQubitWeylDecomposition::new_inner(unitary, Some(DEFAULT_FIDELITY), None)?;
+
+        let euler_basis = EulerBasis::ZYZ;
+        let mut target_1q_basis_list = EulerBasisSet::new();
+        target_1q_basis_list.add_basis(euler_basis);
+
+        let c1r = target_decomposed.K1r.view();
+        let c2r = target_decomposed.K2r.view();
+        let c1l = target_decomposed.K1l.view();
+        let c2l = target_decomposed.K2l.view();
+
+        let unitary_c1r =
+            unitary_to_gate_sequence_inner(c1r, &target_1q_basis_list, 0, None, true, None);
+        let unitary_c2r =
+            unitary_to_gate_sequence_inner(c2r, &target_1q_basis_list, 0, None, true, None);
+        let unitary_c1l =
+            unitary_to_gate_sequence_inner(c1l, &target_1q_basis_list, 0, None, true, None);
+        let unitary_c2l =
+            unitary_to_gate_sequence_inner(c2l, &target_1q_basis_list, 0, None, true, None);
+
+        let mut gates = Vec::new();
```

Comment:
Same comment here about pre-allocating:
```suggestion
        let mut gates = Vec::with_capacity(59);
```
I counted 47 max entries for `weyl_gate()` calls plus 12 (4x 3 gates) for each 1q component.

Reply:
Done in [1ceb94e](https://github.com/Qiskit/qiskit/pull/13139/commits/1ceb94e13e9f3abee67736b19017dc3063fdffb2)